libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKUpgradeManager.java

```diff
@@ -68,6 +72,25 @@ public static synchronized SalesforceSDKUpgradeManager getInstance() {
         return INSTANCE;
     }
 
+    /**
+     * Simple interface to improve testability
+     */
+    interface UserManager {
```

Comment:
To decouple this class from UserAccountManager and make testing easier.
The default constructor (no params) creates a UserManager that is wired to `SalesforceSDKManager.getInstance().getUserAccountManager().getAuthenticatedUsers()` but in tests we can pass a simpler UserManager.